### DISCUSSION TITLE
Need help in drawing the shapes like wave,arrow on image.

### DISCUSSION THREAD

**Discussion by gmek5476** [ImageMagick/ImageMagick#6485] [Development]
### ImageMagick version

7.1.1-7

### Operating system, version and so on

linux and windows

### Description

I have a requirement to draw shape on images and fill with color.The shapes are different line wave, arc and arrow.Please find the attachment for final expected images.i thought to go with drawing polygon but getting the white border on leftside is a challenge to me.Any help is appreciated.
![screen1](https://github.com/ImageMagick/ImageMagick/assets/126016862/5a472486-6753-482e-8e0f-ebc550122936)
![screen2](https://github.com/ImageMagick/ImageMagick/assets/126016862/1e130c9d-75d6-483f-aaee-19c13dc550fd)
![screen3](https://github.com/ImageMagick/ImageMagick/assets/126016862/4766935d-612d-44f0-bf23-a2e996eaa1bb)
![screen4](https://github.com/ImageMagick/ImageMagick/assets/126016862/28b3419c-a08d-415c-bd6e-c1894302ef31)

In the attached images i have image on right side and filled with color on left side, Please ignore the text on images.

**Comment by snibgo** [comment]
I'm not sure what you are asking. Please show a sample input image, and samples of what outputs you want from the input.

**Comment by gmek5476** [comment]
Hi Snigbo ,
Please find the attached for source and target

**Comment by snibgo** [comment]
This does what I think you want. I have used `Orange` for the colour on the left, and `Pink` for the line. Change those colours to whatever you want. The line is a Bezier curve. Change that to whatever you want.

Every line that writes to x*.png, like "+write x1.png", is for testing and debugging. You can remove those when it does what you want.

Windows BAT syntax.
```
magick ^
  gmekBase.JPG ^
  +write mpr:INP ^
  +delete ^
  ( mpr:INP ^
    -fill Black -colorize 100 ^
    -fill None -stroke White -strokewidth 20 ^
    -draw "Bezier 470,-10 270,%%[fx:h/3] 570,%%[fx:h*2/3] 470,%%[fx:h+10]" ^
    +write mpr:LINE ^
    +write x1.png ^
    +delete ^
  ) ^
  ( mpr:LINE ^
    -fill White ^
    -bordercolor White ^
    -draw "color 0,0 filltoborder" ^
    -fill Black +opaque White ^
    +write x2.png ^
    +write mpr:SEMITRANS ^
    +delete ^
  ) ^
  mpr:INP ^
  -fill Orange -colorize 100 ^
  mpr:SEMITRANS ^
  -alpha off -compose CopyOpacity -composite ^
  mpr:INP ^
  -compose DstOver -composite ^
  +write x3.png ^
  ( mpr:LINE -alpha Copy -fill Pink -colorize 100 ^
    +write x4.png ^
  ) ^
  -compose Over -composite ^
  out.png
```

![out](https://github.com/ImageMagick/ImageMagick/assets/33812368/8c2cf385-e88e-434f-b24c-28b10b348926)

**Comment by gmek5476** [comment]
Awesome snibgo,
Can I apply opacity for the color orange?

Thanks
Govind

```
From: snibgo ***@***.***>
Sent: Thursday, July 13, 2023 11:21 AM
To: ImageMagick/ImageMagick ***@***.***>
Cc: Mekala, Govinda Raju ***@***.***>; Author ***@***.***>
Subject: Re: [ImageMagick/ImageMagick] Need help in drawing the shapes like wave,arrow on image. (Discussion #6485)

EXTERNAL EMAIL: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe

This does what I think you want. I have used Orange for the colour on the left, and Pink for the line. Change those colours to whatever you want. The line is a Bezier curve. Change that to whatever you want.

Every line that writes to x*.png, like "+write x1.png", is for testing and debugging. You can remove those when it does what you want.

Windows BAT syntax.

magick ^

  gmekBase.JPG ^

  +write mpr:INP ^

  +delete ^

  ( mpr:INP ^

    -fill Black -colorize 100 ^

    -fill None -stroke White -strokewidth 20 ^

    -draw "Bezier 470,-10 270,%%[fx:h/3] 570,%%[fx:h*2/3] 470,%%[fx:h+10]" ^

    +write mpr:LINE ^

    +write x1.png ^

    +delete ^

  ) ^

  ( mpr:LINE ^

    -fill White ^

    -bordercolor White ^

    -draw "color 0,0 filltoborder" ^

    -fill Black +opaque White ^

    +write x2.png ^

    +write mpr:SEMITRANS ^

    +delete ^

  ) ^

  mpr:INP ^

  -fill Orange -colorize 100 ^

  mpr:SEMITRANS ^

  -alpha off -compose CopyOpacity -composite ^

  mpr:INP ^

  -compose DstOver -composite ^

  +write x3.png ^

  ( mpr:LINE -alpha Copy -fill Pink -colorize 100 ^

    +write x4.png ^

  ) ^

  -compose Over -composite ^

  out.png

[out]<https://urldefense.com/v3/__https:/user-images.githubusercontent.com/33812368/253342477-8c2cf385-e88e-434f-b24c-28b10b348926.png__;!!K-0RPh4!_tkEBnu8-fCol7i-t5OeX42ftbDVxnY0tAFrgUkX8ZUSG8mOQfrpfOXJNh040wMFTiDGcfiPcPIKoSg0o-nLDUir3AMA7CI$>

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/ImageMagick/ImageMagick/discussions/6485*discussioncomment-6441209__;Iw!!K-0RPh4!_tkEBnu8-fCol7i-t5OeX42ftbDVxnY0tAFrgUkX8ZUSG8mOQfrpfOXJNh040wMFTiDGcfiPcPIKoSg0o-nLDUirIuS99DI$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/A6BN2XSBAR4PDNM5EFVFQZLXQAOATANCNFSM6AAAAAA2JAFLVE__;!!K-0RPh4!_tkEBnu8-fCol7i-t5OeX42ftbDVxnY0tAFrgUkX8ZUSG8mOQfrpfOXJNh040wMFTiDGcfiPcPIKoSg0o-nLDUirCBGUyIg$>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>
```

**Comment by snibgo** [comment]
> Can I apply opacity for the color orange?

I don't know what effect you want. I offer two alternative effects.

**Effect A:**

You can do that by adjusting opacity (by multiplying the alpha channel). Insert a line "`-channel A -evaluate Multiply 0.25 +channel ^`" so part of the command is:
```
  -fill Orange -colorize 100 ^
  mpr:SEMITRANS ^
  -alpha off -compose CopyOpacity -composite ^
  -channel A -evaluate Multiply 0.25 +channel ^
  mpr:INP ^
  -compose DstOver -composite ^
```
However, the semi-transparent orange is then composited over the input image, which is gray in that area, so the orange is dulled (made grayer) and made opaque. The output image is fully opaque.

**Effect B:**

In this effect, the output image has the orange area semi-transparent, and pure orange. To get this, instead of compositing over `mpr:INP`, we first make that part of the input image transparent. So those lines become:
```
  -fill Orange -colorize 100 ^
  mpr:SEMITRANS ^
  -alpha off -compose CopyOpacity -composite ^
  -channel A -evaluate Multiply 0.25 +channel ^
  ( mpr:SEMITRANS -negate mpr:INP +swap -alpha off -compose CopyOpacity -composite ) ^
  -compose DstOver -composite ^
```
